### ISSUE TITLE
Memory leak issue with Imgui component

### ISSUE THREAD

**Issue by Cai1Hsu** [Cai1Hsu/DanmakuEngine#66]
```
24-03-29 08:02:52.369 [Error]: [Thread-Update] Unhandled exception: Insufficient memory to continue the execution of the program.
   at System.Runtime.InteropServices.Marshal.AllocHGlobal(IntPtr cb)
   at DanmakuEngine.DearImgui.Graphics.ImguiDrawDataSnapshot.WriteNewFrame(ImDrawDataPtr src) in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Graphics\ImguiDrawDataSnapshot.cs:line 64
   at DanmakuEngine.DearImgui.Imgui.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Imgui_Frame.cs:line 33
   at DanmakuEngine.Engine.GameHost.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Engine\GameHost.cs:line 204
   at DanmakuEngine.Scheduling.FrameExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\FrameExecutor.cs:line 40
   at DanmakuEngine.Scheduling.ThrottledExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\ThrottledExecutor.cs:line 33
   at DanmakuEngine.Threading.GameThread.processFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Threading\GameThread.cs:line 377
24-03-29 08:02:52.382 [Error]: [Thread-Update] Unhandled exception: Insufficient memory to continue the execution of the program.
   at System.Runtime.InteropServices.Marshal.AllocHGlobal(IntPtr cb)
   at DanmakuEngine.DearImgui.Graphics.ImguiDrawDataSnapshot.WriteNewFrame(ImDrawDataPtr src) in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Graphics\ImguiDrawDataSnapshot.cs:line 64
   at DanmakuEngine.DearImgui.Imgui.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Imgui_Frame.cs:line 33
   at DanmakuEngine.Engine.GameHost.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Engine\GameHost.cs:line 204
   at DanmakuEngine.Scheduling.FrameExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\FrameExecutor.cs:line 40
   at DanmakuEngine.Scheduling.ThrottledExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\ThrottledExecutor.cs:line 33
   at DanmakuEngine.Threading.GameThread.processFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Threading\GameThread.cs:line 377
24-03-29 08:02:52.395 [Error]: [Thread-Update] Unhandled exception: Insufficient memory to continue the execution of the program.
   at System.Runtime.InteropServices.Marshal.AllocHGlobal(IntPtr cb)
   at DanmakuEngine.DearImgui.Graphics.ImguiDrawDataSnapshot.WriteNewFrame(ImDrawDataPtr src) in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Graphics\ImguiDrawDataSnapshot.cs:line 64
   at DanmakuEngine.DearImgui.Imgui.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\DearImgui\Imgui_Frame.cs:line 33
   at DanmakuEngine.Engine.GameHost.Update() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Engine\GameHost.cs:line 204
   at DanmakuEngine.Scheduling.FrameExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\FrameExecutor.cs:line 40
   at DanmakuEngine.Scheduling.ThrottledExecutor.RunFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Scheduling\ThrottledExecutor.cs:line 33
   at DanmakuEngine.Threading.GameThread.processFrame() in C:\Users\Caiyi\Desktop\DI\DanmakuEngine\DanmakuEngine\Threading\GameThread.cs:line 377
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Repeat 2 times:
--------------------------------
   at ImGuiNET.ImGuiNative.igShowDemoWindow(Byte*)   at ImGuiNET.ImGui.ShowDemoWindow()
   at DanmakuEngine.DearImgui.Windowing.ImguiDemoWindow.Update()
   at DanmakuEngine.DearImgui.Windowing.ImguiWindowBase.UpdateUI()
   at DanmakuEngine.DearImgui.Windowing.ImguiWindowManager.UpdateWindows()
   at DanmakuEngine.DearImgui.Imgui.Update()   at DanmakuEngine.Engine.GameHost.Update()   at DanmakuEngine.Scheduling.FrameExecutor.RunFrame()
   at DanmakuEngine.Scheduling.ThrottledExecutor.RunFrame()   at DanmakuEngine.Threading.GameThread.processFrame()   at DanmakuEngine.Threading.GameThread.RunSingleFrame()   at DanmakuEngine.Threading.GameThread.<createThread>g__runWork|53_0()
```

**Comment by Cai1Hsu** [comment]
Fixed in [2c9da05](https://github.com/Cai1Hsu/DanmakuEngine/commit/2c9da05a27afa00dcd8d2ae18b27fd14082c1c41)

**Comment by Cai1Hsu** [comment]
Multithread Dear imGui support should be stable now
